src/table/body-cell/index.tsx

```diff
@@ -85,12 +102,17 @@ function TableCellEditable<ItemType>({
       ) : (
         <>
           {column.cell(item)}
-          {successfulEdit && (
+          {showSuccessIcon && showIcon && (
             <>
               <span
                 className={styles['body-cell-success']}
                 aria-label={ariaLabels?.successfulEditLabel?.(column)}
                 role="img"
+                onMouseDown={e => {
+                  // Prevent the editor's Button blur event to be fired when clicking the success icon.
+                  // This prevents unfocusing the button and triggers the `TableTdElement` onClick event which initiates the edit mode.
+                  e.preventDefault();
```

Comment:
Could this have been solved with `pointer-events` on the icon? That might be a bit more "elegant". If not, this is fine too.

Reply:
I've tried that first but a `pointer-events: none` would results in a little flickering from focus state to hover state to edit state when clicking the success icon: `onBlur` gets triggered before `onClick` where Button's `onBlur` removes the focus state followed by the `onClick` which enables the edit state:

https://github.com/cloudscape-design/components/assets/569011/16ad87b0-e821-4f2a-a898-c2251a380abf

Intercepting the click on `onMouseDown` prevents `onBlur` from being called. This gets the cell directly into edit mode.